app/jenkins-for-jira-ui/src/components/ConnectJenkins/CreateServer/CreateServer.tsx

```diff
@@ -0,0 +1,91 @@
+import React, { useEffect, useState } from 'react';
+import Button, { LoadingButton } from '@atlaskit/button';
+import Form, {
+	FormFooter
+} from '@atlaskit/form';
+import { v4 as uuidv4 } from 'uuid';
+import { useHistory } from 'react-router';
+import {
+	loadingIcon
+} from '../../JenkinsConfigurationForm/JenkinsConfigurationForm.styles';
+import { createJenkinsServer } from '../../../api/createJenkinsServer';
+import { ConfigurationSteps } from '../ConfigurationSteps/ConfigurationSteps';
+import { StyledH1, StyledInstallationContainer, StyledInstallationContent } from '../ConnectJenkins.styles';
+import { generateNewSecret } from '../../JenkinsConfigurationForm/JenkinsConfigurationForm';
+import { isFormValid, setName } from '../../../common/util/jenkinsConnectionsUtils';
+import { ServerConfigurationFormName } from '../../JenkinsConfigurationForm/ServerConfigurationFormElements/ServerConfigurationFormName/ServerConfigurationFormName';
+import { ConnectLogos } from '../ConnectLogos/ConnectLogos';
+
+const CreateServer = () => {
+	const history = useHistory();
+	const [uuid] = useState(uuidv4);
+	const [serverName, setServerName] = useState('');
+	const [secret, setSecret] = useState('');
+	const [hasError, setHasError] = useState(false);
+	const [isLoading, setIsLoading] = useState(false);
+	const [errorMessage, setErrorMessage] = useState('');
+
+	useEffect(() => {
+		setSecret(generateNewSecret());
+	}, [uuid]);
+
+	const createServer = async () => {
+		const isValidForm = isFormValid(serverName, setHasError, setErrorMessage);
+
+		if (isValidForm) {
+			setIsLoading(true);
+
+			try {
+				await createJenkinsServer({
+					name: serverName,
+					uuid,
+					secret,
+					pipelines: []
+				});
```

Comment:
What happens when this fails? Remember that all frontend validations can be circumvented.

Reply:
We currently have a catch that console logs an error. Unfortunately, we haven't set up proper error handling for the front end of this app. We are relying on the API for this. This is something we should look into, so lets investigate and address this in a separate ticket.

Reply:
Ticket: https://softwareteams.atlassian.net/browse/ARC-1554